homeassistant/components/tado/__init__.py

```diff
@@ -172,6 +172,14 @@ def setup(self):
         self.home_id = tado_home["id"]
         self.home_name = tado_home["name"]
 
+    def getMe(self):
```

Comment:
You're always calling this one in `self.hass.async_add_executor_job`, this object also knows `hass`, can we maybe make this function async and do the executor job here? This way you can make the other code cleaner

Reply:
`getMobileDevices` as well

Reply:
My two cents would be that this would strafe away from how `__init__.py` has been setup, as a main class. The reason I need to use `self.hass.async_add_executor_job` is because of the older DeviceScanner class. Once we make a PR to update it to the ScannerEntity, in combination with a DataUpdateCoordinator the `self.hass.async_add_executor_job` is redundant.
PyTado is no async compatible, unfortunally.

Reply:
I know, but what I am proposing is moving away from `async_add_executor_job(self.tadoconnector.getMe)` to `await self.tadoconnector.getMe()` by changing this implementation